### PR TITLE
Removed Filter should reappear without selection

### PR DESCRIPTION
**Pull request by BenediktSeidl**
* add host filter
* choose host "X"
* remove host filter
* add host filter

-> host "X" was selected, but graph did not reflect this

Now the selection of the host filter (and other filters) is empty when
readding them.

### CODE REVIEW

File: src/components/combinedgraphs.tsx [hunk 1]
```diff
@@ -86,6 +86,9 @@ export const SelectFilters = (props: FilterEditorProps): null | JSX.Element => {
     if (query.context) {
       delete query.context[filtername];
     }
+    if (query.params && query.params.selections && query.params.selections.context) {
```
Comment (PhilippSiegmantel) [inline src/components/combinedgraphs.tsx:89]: This looks like an opportunity to use the ".?" operator.

File: src/types.ts [hunk 2]
```diff
@@ -6,10 +6,15 @@ export type Context = Record<string, ContextHTTPVars>;
 
 export type Presentation = 'lines' | 'sum' | 'average' | 'min' | 'max';
 
+export interface Selections {
```
Comment (PhilippSiegmantel) [inline src/types.ts:9]: Do we need this Interface somewhere else? If not, I'd inline it.

File: src/types.ts [hunk 3]
```diff
@@ -6,10 +6,15 @@ export type Context = Record<string, ContextHTTPVars>;
...
+  context?: Record<string, unknown>;
+}
+
 export interface QueryParams {
```
Comment (PhilippSiegmantel) [inline src/types.ts:13]: Not your fault, but an Interface where everything is optional makes me nervous. Do we know when we need what? Or if some things have defaults?

**Review by PhilippSiegmantel — CHANGES_REQUESTED** [review]
I'd love to see a test for this as well.

File: src/types.ts (hunk 3 above)
Comment (BenediktSeidl) [reply src/types.ts:13]: To be honest it's just defensive programming. As we didn't have a typing error without this, we basically don't know anything about it. This is why I choose to make it optional. I think we will have to have some refactoring in order to be sure about this?!

File: src/components/combinedgraphs.tsx (hunk 1 above)
Comment (BenediktSeidl) [reply src/components/combinedgraphs.tsx:89]: This is embarrassing…

**Review by PhilippSiegmantel — APPROVED** [review]

**Comment by BenediktSeidl** [comment]
the bug will be fixed with #102 so we don't need to merge this.

**Pull request closed without merging**